src/Surfnet/StepupSelfService/SelfServiceBundle/Tests/RemoteVetting/Service/IdentityEncrypterIntegrationTest.php

```diff
@@ -0,0 +1,170 @@
+<?php
+
+/**
+ * Copyright 2020 SURFnet B.V.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+namespace Surfnet\StepupSelfService\SelfServiceBundle\Tests\RemoteVetting\Service;
+
+use Exception;
+use Mockery as m;
+use PHPUnit_Framework_TestCase as UnitTest;
+use Surfnet\StepupSelfService\SelfServiceBundle\RemoteVetting\Configuration\RemoteVettingConfiguration;
+use Surfnet\StepupSelfService\SelfServiceBundle\RemoteVetting\Dto\IdentityDto;
+use Surfnet\StepupSelfService\SelfServiceBundle\RemoteVetting\Service\IdentityEncrypter;
+use Surfnet\StepupSelfService\SelfServiceBundle\RemoteVetting\Service\IdentityFilesystemWriter;
+use Surfnet\StepupSelfService\SelfServiceBundle\RemoteVetting\Service\IdentityWriterInterface;
+
+/**
+ * By using a fake IdentityWriter we are able to intercept the encrypted data (that would otherwise
+ * be stored to a backend) and perform some sanity checks on it.
+ */
+class IdentityEncrypterIntegrationTest extends UnitTest
+{
+    private $encrypter;
+
+    private $config;
+
+    private $writer;
+
+    private $publicKey = <<<CERT
```

Comment:
Could we use files for the keys?

Reply:
We sure could, it might be cleaner from a code perspective.